astericks/astericks.c++

```diff
@@ -0,0 +1,28 @@
+#include <iostream>
+
+using namespace std;
+
+int main(){
+
+    int rows{};
+    int cols{};
+    cout<<"Number of rows: ";
+    cin>> rows;
+    cout<<"Number of cols: ";
+    cin>>cols;
+
+    for (int i{1}; i <= rows ; i++){
+        for(int j{1}; j <= cols ; j++){
+            if( i == 1 || i == rows || j==1 || j == cols){
+                std::cout<<"*";
+            }
+            else{
+                std::cout<<" ";
+            };
+        };
+        std::cout<<'\n';

```

Comment:
why not std:endl?

Reply:
As you know std::cout is buffered meaning that the request is stored in a region of memory and is flushed periodically.
std::endl moves the cursor to the next line of the console, and it flushes the buffer. it's more efficient to let the system do the flushing periodically. So to prevent that kind of behavior I used " std::cout<<'\n'; "